ocpp/messages.py

```diff
@@ -134,6 +141,55 @@ def get_schema(message_type_id, action, ocpp_version, parse_float=float):
     return _schemas[relative_path]
 
 
+def get_validator(
+        message_type_id: int,
+        action: str,
+        ocpp_version: str,
+        parse_float: Callable = float
+) -> Draft4Validator:
+    """
+    Read schema from disk and return as `Draft4Validator`. Instances will be
+    cached for performance reasons.
+
+    The `parse_float` argument can be used to set the conversion method that
+    is used to parse floats. It must be a callable taking 1 argument. By
+    default it is `float()`, but certain schema's require `decimal.Decimal()`.
+    """
+    if ocpp_version not in ["1.6", "2.0", "2.0.1"]:
+        raise ValueError
+
+    schemas_dir = 'v' + ocpp_version.replace('.', '')
+
+    schema_name = action
+    if message_type_id == MessageType.CallResult:
+        schema_name += 'Response'
+    elif message_type_id == MessageType.Call:
+        if ocpp_version in ["2.0", "2.0.1"]:
+            schema_name += 'Request'
+
+    if ocpp_version == "2.0":
+        schema_name += '_v1p0'
+
+    cache_key = schema_name + ocpp_version
```

Comment:
One small suggestion for readability/consistency would be to put an underscore or something to separate the name and version. Not a big deal either way though.

Reply:
Thanks for the feedback. I changed it.